File: src/main/java/de/kevcodez/ecommerce/parser/domain/image/Image.java
package de.kevcodez.ecommerce.parser.domain.image;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

import lombok.Data;

@Data
public class Image {

    private final List<ImageVariant> variants = new ArrayList<>();

    public Image() {
    }

    public Image addVariant(ImageVariant imageVariant) {
        variants.add(imageVariant);
        return this;
    }

    public List<ImageVariant> getVariants() {
        return Collections.unmodifiableList(variants);
    }

}

File: src/main/kotlin/de/kevcodez/ecommerce/parser/domain/image/Image.kt
package de.kevcodez.ecommerce.parser.domain.image

data class Image(val variants: ArrayList<ImageVariant>) {

    constructor() : this(ArrayList())

    fun addVariant(imageVariant: ImageVariant): Image {
        this.variants.add(imageVariant)
        return this
    }

}
